pkg/api/types.go

```diff
@@ -2573,6 +2573,31 @@ const (
 	ServiceAffinityNone ServiceAffinity = "None"
 )
 
+const (
+	// ServiceAffinityClientIPDefaultSeconds is the default timeout seconds
+	// of Client IP based session affinity - 3 hours.
+	ServiceAffinityClientIPDefaultSeconds int32 = 10800
+	// ServiceAffinityClientIPMaxAgeSeconds is the max timeout seconds
+	// of Client IP based session affinity - 1 day.
+	ServiceAffinityClientIPMaxAgeSeconds int32 = 86400
+)
+
+// ServiceAffinityConfig represents the configurations of session affinity.
+type ServiceAffinityConfig struct {
+	// ClientIPConfig contains the configurations of Client IP based session affinity.
+	// +optional
+	ClientIPConfig *ServiceAffinityConfigClientIP
+}
+
+// ServiceAffinityConfigClientIP represents the configurations of Client IP based session affinity.
+type ServiceAffinityConfigClientIP struct {
+	// TimeoutSeconds specifies the seconds of ClientIP type session sticky time.
+	// The value must be >0 && <=86400(for 1 day) if ServiceAffinity == "ClientIP".
+	// Default value is 86400 for 1 day.
+	// +optional
```

Comment:
Thanks for your good example.

Currently, I changed `TimeoutSeconds` to be a required field when clientIPConfig is not nil.